secure/storage/src/namespaced_storage.rs

```diff
@@ -1,29 +1,61 @@
 // Copyright (c) The Diem Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
-use crate::{CryptoKVStorage, Error, GetResponse, KVStorage, Storage};
+use crate::{CryptoStorage, Error, GetResponse, KVStorage, PublicKeyResponse};
+use diem_crypto::{
+    ed25519::{Ed25519PrivateKey, Ed25519PublicKey, Ed25519Signature},
+    hash::CryptoHash,
+};
 use serde::{de::DeserializeOwned, Serialize};
 
 /// This provides a light wrapper around KV storages to support a namespace. That namespace is
 /// effectively prefixing all keys with then namespace value and "/" so a namespace of foo and a
-/// key of bar becomes "foo/bar". Without a namespace, the key would just be "bar". This matches
-/// how this library implements namespaces for Vault.
-pub struct NamespacedStorage {
+/// key of bar becomes "foo/bar". Without a namespace, the key would just be "bar".
+pub struct Namespaced<S> {
     namespace: String,
-    inner: Box<Storage>,
+    inner: S,
```

Comment:
Why does vault need to be boxed?

Reply:
This line is removed the requirement to have a namespaced storage boxed so i'm not sure what you're asking